File: docs/guides/dart/lib/saveObjectsMCM.dart
import 'package:algolia_client_search/algolia_client_search.dart';

List<Map<String, String>> getAllAppIDConfigurations() {
  return [/* A list of your MCM AppID/ApiKey pairs */];
}

List<Map> playlists = [/* Your records */];

void saveObjectsMCM() async {
  final configurations = getAllAppIDConfigurations();

  for (var configuration in configurations) {
    final appId = configuration['appID'] ?? "";
    final apiKey = configuration['apiKey'] ?? "";
    final client = SearchClient(appId: appId, apiKey: apiKey);

    try {
      final batchParams = BatchWriteParams(
          requests: playlists
              .map((record) => BatchRequest(
                    action: Action.addObject,
                    body: record,
                  ))
              .toList());
      await client.batch(
        indexName: "<YOUR_INDEX_NAME>",
        batchWriteParams: batchParams,
      );
    } catch (e) {
      throw Exception('Error for appID $appId: $e');
    }
  }
}

File: docs/guides/scala/src/main/scala/saveObjectsMCM.scala
import scala.concurrent.Future
import scala.concurrent.ExecutionContext.Implicits.global

import algoliasearch.api.SearchClient
import algoliasearch.config.*
import algoliasearch.extension.SearchClientExtensions
import algoliasearch.config.RequestOptions

val getAllAppIDConfigurations: () => Map[String, String] = () => {
  Map( /* A map of your MCM AppID/ApiKey pairs */ )
}

val playlists: Seq[Any] = Seq( /* Your records */ )

def saveObjectsMCM(): Future[Unit] = {
  val configurations = getAllAppIDConfigurations()

  Future
    .sequence {
      configurations.map { case (appID, apiKey) =>
        val client = new SearchClient(appID, apiKey)

        client
          .saveObjects(
            indexName = "<YOUR_INDEX_NAME>",
            objects = playlists
          )
          .recover { case ex: Exception =>
            println(s"Error for appID $appID: ${ex.getMessage}")
          }
      }
    }
    .map(_ => ())
}
